rotkehlchen/tasks/manager.py

```diff
@@ -397,25 +401,38 @@ def _maybe_decode_evm_transactions(self) -> None:
     def _maybe_check_premium_status(self) -> None:
         """
         Validates the premium status of the account and if the credentials are not valid
-        it deactivates the user's premium status.
+        it retries for 3 times before deactivating the user's premium status, but if the
+        credentials are valid it will reactivate the user's premium status.
         """
         now = ts_now()
         if now - self.last_premium_status_check < PREMIUM_STATUS_CHECK:
             return
 
         db_credentials = self.database.get_rotkehlchen_premium()
-        if db_credentials:
-            try:
-                premium_create_and_verify(db_credentials)
-            except PremiumAuthenticationError as e:
-                message = (
-                    f'Could not authenticate with the rotkehlchen server with '
-                    f'the API keys found in the Database. Error: {str(e)}. Will '
-                    f'deactivate the premium status.'
-                )
-                self.msg_aggregator.add_error(message)
-                self.deactivate_premium()
-        self.last_premium_status_check = now
+        if not db_credentials:
+            self.last_premium_status_check = now
+            return
+
+        try:
+            premium_create_and_verify(db_credentials)
+        except PremiumAuthenticationError as e:
+            if not self.premium_check_retries >= PREMIUM_CHECK_RETRY_LIMIT:
```

Comment:
Adding negations to logic clauses always introduces mental complexity to the code since is difficult for humans to follow it. Could you just change the condition?

```suggestion
            if self.premium_check_retries <= PREMIUM_CHECK_RETRY_LIMIT:
```